services/alertmanager/registry.go

```diff
@@ -76,14 +70,14 @@ func (r *registry) RemovePrefix(prefix string) {
 }
 
 // Collect returns all firing alerts.
-func (r *registry) Collect() ammodels.PostableAlerts {
+func (r *Registry) Collect() ammodels.PostableAlerts {
 	r.rw.RLock()
 	defer r.rw.RUnlock()
 
 	var res ammodels.PostableAlerts
 	now := time.Now()
 	for id, t := range r.times {
-		if t.After(now) {
+		if t.Before(now) {
```

Comment:
Love it)

Reply:
LOL. I remember agonising on that line, changing it several times.

@atymchuk that explains disappearing alerts

Reply:
Cool, then the fix is not insanely difficult :)